core/storage/ipfs/merkledag/impl/node_impl.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/ipfs/merkledag/impl/node_impl.hpp"
+
+#include <libp2p/crypto/sha/sha256.hpp>
+#include <libp2p/multi/content_identifier_codec.hpp>
+#include <libp2p/multi/multibase_codec/codecs/base58.hpp>
+#include <libp2p/multi/multihash.hpp>
+#include "merkledag.pb.h"
+#include "storage/ipfs/merkledag/impl/pb_node_decoder.hpp"
+
+using libp2p::common::Hash256;
+using libp2p::multi::ContentIdentifierCodec;
+using libp2p::multi::HashType;
+using libp2p::multi::MulticodecType;
+using libp2p::multi::Multihash;
+using merkledag::pb::PBLink;
+using merkledag::pb::PBNode;
+using Version = libp2p::multi::ContentIdentifier::Version;
+
+namespace fc::storage::ipfs::merkledag {
+
+  size_t NodeImpl::size() const {
+    return getCachePB().size() + child_nodes_size_;
+  }
+
+  void NodeImpl::assign(common::Buffer input) {
+    content_ = std::move(input);
+    pb_cache_ = boost::none;
+  }
+
+  const common::Buffer &NodeImpl::content() const {
+    return content_;
+  }
+
+  outcome::result<void> NodeImpl::addChild(const std::string &name,
+                                           std::shared_ptr<const Node> node) {
+    LinkImpl link{node->getCID(), name, node->size()};
+    std::pair<std::string, LinkImpl> data{name, std::move(link)};
```

Comment:
try to use std::make_pair() to simplify

Reply:
Simplified to `links_.emplace(name, std::move(link));` instead `links_.emplace(std::make_pair...)`